client/static/js/board.js

```diff
@@ -6,188 +6,118 @@ var Board = {
-        ctx.translate(this.x, this.y);
-        ctx.fillStyle = this.color;
-        ctx.fillRect(this.x*100 -4, this.y*100-4,(this.x2-this.x)*100+4,(this.y2-this.y)*100+4);
-        ctx.restore();
+    // edge = {src: {x: int, y: int}, dst: {x: int, y: int}, owner: String}
+    claimEdge: function(x1, y1, x2, y2, color) {
+        // TODO: Proper edge claiming
+        // If valid, render and edge and notify server via websockets
+        this.edges.push(new Edge(x1, y1, x2, y2, color));
     }
-}
-
-var camera = {
-    x: 0,
-    y: 0,
-    width: 1000,
-    height: 600
 };
 
-Board.canvas.addEventListener('mousedown',function(e) {
-    mousex= e.pageX;
-    mousey= e.pageY;
-    //get closest point
-    mousex-=camera.x;
-    mousey-=camera.y;
-    pointx = Math.round(mousex/100);
-    pointy = Math.round(mousey/100);
-    pointx2 = pointx;
-    pointy2 = pointy;
-
-    //get second point
-    if ((mousex%100)>=(mousey%100) && mousex%100 <50 && mousey%100 <50) {
-        pointx2 = parseInt(pointx)+1;
-    }
-    else if ((mousex%100)<(mousey%100) && mousex%100 <50 && mousey%100 <50){
-        pointy2 = parseInt(pointy)+1;
+function updateBoard(timestamp) {
+    if (timestamp < Board.lastFrameTimeMs + (1000 / Board.maxFPS)) {
+            requestAnimationFrame(updateBoard);
+            return;
     }
 
-    if ((mousex%100)>=(mousey%100) && mousex%100 >=50 && mousey%100 <50){
-        pointx2 = parseInt(pointx)-1;
-    }
-    else if ((mousex%100)<(mousey%100) && mousex%100 >=50 && mousey%100 <50){
-        pointy2 = parseInt(pointy)+1;
-    }
+    Board.lastFrameTimeMs = timestamp;
+    Board.clear();
+    Board.cursor.update();
 
-    if ((mousex%100)>=(mousey%100) && mousex%100 <50 && mousey%100 >=50){
-        pointx2 = parseInt(pointx)-1;
-    }
-    else if ((mousex%100)<(mousey%100) && mousex%100 <50 && mousey%100 >=50) {
-        pointy2 = parseInt(pointy)-1;
+    for(var i = 0; i < Board.edges.length; i++){
+        Board.edges[i].update();
     }
 
-    if ((mousex%100)<(mousey%100) && mousex%100 >=50 && mousey%100 >=50){
-        pointx2 = parseInt(pointx)-1;
-    }
-    else if ((mousex%100)>=(mousey%100) && mousex%100 >=50 && mousey%100 >=50) {
-        pointy2 = parseInt(pointy)-1;
-    }
-    Board.edges.push(new edge(pointx,pointy,pointx2,pointy2,'blue'));
-
-});
-
-Board.canvas.addEventListener('mousemove',function(e) {
-    mousex= e.pageX;
-    mousey= e.pageY;
-    Board.curx =mousex;
-    Board.cury = mousey;
-
-    mousex-=camera.x;
-    mousey-=camera.y;
-    pointx = Math.round(mousex/100)
-    pointy = Math.round(mousey/100)
-    pointx2 = pointx;
-    pointy2 = pointy;
-
-    //get second point
-    if ((mousex%100)>=(mousey%100) && mousex%100 <50 && mousey%100 <50) {
-        pointx2 = parseInt(pointx)+1;
-    }
-    else if ((mousex%100)<(mousey%100) && mousex%100 <50 && mousey%100 <50){
-        pointy2 = parseInt(pointy)+1;
+    if (Board.curx > (Board.canvas.width - 100) && Board.camera.x > -3000) {
```

Comment:
I haven't done this, I will leave this to @kaptainkohl (he has a better understanding of the numerical values than I), but I did break this into a helper function since it was being used in two places.